include/swift/AST/DiagnosticsSema.def

```diff
@@ -57,6 +57,11 @@ ERROR(could_not_find_value_subscript,none,
       "value of type %0 has no subscripts",
       (Type))
 
+ERROR(could_not_find_value_subscript_key_path,none,
+      "value of type %0 has no subscripts; "
+      "did you mean to use a key path subscript?",
+      (Type))
```

Comment:
Will this diagnostic be emitted even when the type might have explicit non-keypath subscripts? I can't tell by looking at the code directly; I would guess that it should be (or at least, the "did you mean to use a key path subscript" would still be applicable), and if that's the case, then the phrasing needs to avoid saying "has no subscripts". Not sure how we could phrase it better though...

Reply:
That's a good question, let me see if I can improve this situation :)

Reply:
I think the new message makes more sense, related to cases where we have 1+ subscripts on the base type, there is something else going on I can't still figure it out. Maybe let it for another patch ... let's try to figure out these ones first :)

Reply:
Thanks, the new diagnostic avoids the potentially incorrect verbiage.

Reply:
> related to cases where we have 1+ subscripts on the base type, there is something else going on I can't still figure it out. Maybe let it for another patch 

Could you leave a `TODO` (either near the diagnostic or where it is being emitted right now) about this, so that it is not forgotten?